fl-server/src/handlers.rs

```diff
@@ -126,16 +121,8 @@ pub async fn create_flist_handler(
     let fl_name = docker_image.replace([':', '/'], "-") + ".fl";
     let username_dir = format!("{}/{}", cfg.flist_dir, username);
 
-    match flist_exists(std::path::Path::new(&username_dir), &fl_name).await {
-        Ok(exists) => {
-            if exists {
-                return Err(ResponseError::Conflict("flist already exists".to_string()));
-            }
-        }
-        Err(e) => {
-            log::error!("failed to check flist existence with error {:?}", e);
-            return Err(ResponseError::InternalServerError);
-        }
+    if std::path::Path::new(std::path::Path::new(&username_dir).join(&fl_name).as_path()).exists() {
```

Comment:
Why do you have to use Path::new() here ? I think `std::path::Path::new(&username_dir).join(&fl_name).exists()` should work, no ?